tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/CalDavEventAttendanceRepository.java

```diff
@@ -0,0 +1,182 @@
+
+import com.google.common.collect.Iterators;
+import com.linagora.tmail.dav.DavCalendarObject;
+import com.linagora.tmail.dav.DavClient;
+import com.linagora.tmail.dav.DavUser;
+import com.linagora.tmail.dav.DavUserProvider;
+import com.linagora.tmail.james.jmap.AttendanceStatus;
+import com.linagora.tmail.james.jmap.EventAttendanceRepository;
+import com.linagora.tmail.james.jmap.MessagePartBlobId;
+import com.linagora.tmail.james.jmap.model.CalendarAttendeeField;
+import com.linagora.tmail.james.jmap.model.CalendarEventAttendanceResults;
+import com.linagora.tmail.james.jmap.model.CalendarEventAttendanceResults$;
+import com.linagora.tmail.james.jmap.model.CalendarEventParsed;
+import com.linagora.tmail.james.jmap.model.CalendarEventReplyResults;
+import com.linagora.tmail.james.jmap.model.EventAttendanceStatusEntry;
+import com.linagora.tmail.james.jmap.model.LanguageLocation;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import scala.collection.JavaConverters;
+
+public class CalDavEventAttendanceRepository implements EventAttendanceRepository {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CalDavEventAttendanceRepository.class);
+    private static final String X_MEETING_UID_HEADER = "X-MEETING-UID";
+
+    private final DavClient davClient;
+    private final SessionProvider sessionProvider;
+    private final MessageId.Factory messageIdFactory;
+    private final MessageIdManager messageIdManager;
+    private final DavUserProvider davUserProvider;
+
+    @Inject
+    public CalDavEventAttendanceRepository(DavClient davClient,
+                                           SessionProvider sessionProvider, MessageId.Factory messageIdFactory,
+                                           MessageIdManager messageIdManager,
+                                           DavUserProvider davUserProvider) {
+        this.davClient = davClient;
+        this.sessionProvider = sessionProvider;
+        this.messageIdFactory = messageIdFactory;
+        this.messageIdManager = messageIdManager;
+        this.davUserProvider = davUserProvider;
+    }
+
+
+    @Override
+    public Publisher<CalendarEventAttendanceResults> getAttendanceStatus(Username username,
+                                                                         BlobIds calendarEventBlobIds) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return Flux.fromIterable(JavaConverters.seqAsJavaList(calendarEventBlobIds.value()))
+            .flatMap(blobId -> getEventUid(blobId.value(), systemMailboxSession)
+                .flatMap(eventUid ->
+                    toDavUser(username)
+                        .flatMap(davUser -> davClient.getCalendarObjectContainingVEvent(davUser, eventUid)))
+                .map(calendarObject -> getAttendanceStatusFromCalendarObject(
+                    BlobId.of(blobId.value()).get(), calendarObject, username, systemMailboxSession)))
```

Comment:
We are converting a blobId into a blobId ? THis looks weird!

Reply:
It is somehow related to interoperability between Scala and Java, especially regarding refined types. I can improve it by defining specialized methods that perform the conversion.